### PR TITLE
feat(web): search filter form

### PR DESCRIPTION
**Pull request by alextran1502**
Dark

Light

### CODE REVIEW

File: web/src/lib/components/shared-components/combobox.svelte [hunk 1]
```diff
@@ -2,27 +2,37 @@
   // Necessary for eslint
   /* eslint-disable @typescript-eslint/no-explicit-any */
   type T = any;
+
+  export type Type = 'button' | 'submit' | 'reset';
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/combobox.svelte:6]: there is probably a provided type for this already. HtmlButtonElement['type'] or something

File: web/src/lib/components/shared-components/search-bar/search-bar.svelte [hunk 2]
```diff
@@ -113,58 +125,20 @@
...
-            Smart search is enabled by default, to search for metadata use the syntax <span
-              class="rounded-lg bg-gray-100 p-2 font-mono font-semibold leading-7 text-immich-primary dark:bg-gray-900 dark:text-immich-dark-primary"
-              >m:your-search-term</span
-            >
-          </p>
-        </div>
+    <!-- SEARCH HISTORY BOX -->
+    {#if showHistory}
+      <SearchHistoryBox
+        on:clearAllSearchTerms={clearAllSearchTerms}
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/search-bar/search-bar.svelte:131]: I would probably prefer an interface of:
```ts
clear: string?
```

Omitting the term implies clear all. And then just have `on:clear={({ detail: term }) => handleClear(term)}`

**Review by jrasm91 — APPROVED** [review]
LGTM

File: web/src/lib/components/shared-components/search-bar/search-bar.svelte (hunk 2 above)
Comment (alextran1502) [reply web/src/lib/components/shared-components/search-bar/search-bar.svelte:131]: One is for clearing the individual search term, one is for clearing all of the terms.

Are you thinking that if we do not pass in the term, we will clear all?

**Review by danieldietzler — APPROVED** [review]
LGTM!

File: web/src/lib/components/shared-components/search-bar/search-bar.svelte (hunk 2 above)
Comment (jrasm91) [reply web/src/lib/components/shared-components/search-bar/search-bar.svelte:131]: Yes

**Pull request merged**